File: src/main/java/org/mifos/fineract/services/SelfRunReport_Api.java
package org.mifos.fineract.services;

import org.mifos.fineract.models.GetRunReportResponse;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import rx.Observable;

public interface SelfRunReport_Api {
    /**
     * Running A Report
     * Example Requests:   self/runreports/Client%20Details?R_officeId&#x3D;1   self/runreports/Client%20Details?R_officeId&#x3D;1&amp;exportCSV&#x3D;true
     *
     * @param reportName reportName (required)
     * @return Observable&lt;GetRunReportResponse&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("self/runreports/{reportName}")
    Observable<GetRunReportResponse> runReport(
            @retrofit2.http.Path("reportName") String reportName
    );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfRunReportApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetRunReportResponse
import retrofit2.http.GET
import retrofit2.http.Path

interface SelfRunReportApi {
    /**
     * Running A Report
     * Example Requests:   self/runreports/Client%20Details?R_officeId&#x3D;1   self/runreports/Client%20Details?R_officeId&#x3D;1&amp;exportCSV&#x3D;true
     * Responses:
     *  - 200: OK
     *
     * @param reportName reportName
     * @return [GetRunReportResponse]
     */
    @GET("v1/self/runreports/{reportName}")
    suspend fun runReport1(@Path("reportName") reportName: String): GetRunReportResponse

}
